### PR TITLE
chore(rpc): migrate more methods to the new framework

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- Cleanup all `LotusJson` wrappers
- Drop `Clone` constraint from `RpcMethod::Params`
- Migrate RPC methods to the new framework (Partially done)

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/rpc/reflect/openrpc_types.rs [hunk 1]
```diff
@@ -144,25 +144,27 @@ impl Params {
...
-            if !late_mandatory_params.is_empty() {
-                return Err(ParamListError(format!(
-                    "Mandatory parameters may not follow optional parameters, \
-                            but the optional parameter {} is followed by [{}]",
-                    first_opt_param.name,
-                    late_mandatory_params.join(", ")
-                )));
-            }
-        };
+
```
Comment (LesnyRumcajs) [inline src/rpc/reflect/openrpc_types.rs:147]: What is this piece of commented-out code?

Comment (hanabi1224) [reply src/rpc/reflect/openrpc_types.rs:147]: I added a comment at the top of the block but it's not obvious. : ) Basically 
this code block panics when mandatory parameters follow optional parameters, but some of the RPC methods violates this rule.

```rust
// `GasEstimateMessageGas::Params` is `(Message, Option<MessageSendSpec>, ApiTipsetKey)
```

Comment (LesnyRumcajs) [reply src/rpc/reflect/openrpc_types.rs:147]: So what do we do with this? In general, commented-out code should not exist.

Comment (hanabi1224) [reply src/rpc/reflect/openrpc_types.rs:147]: I'd create a tracking issue and get input from @aatifsyed

Comment (hanabi1224) [reply src/rpc/reflect/openrpc_types.rs:147]: https://github.com/ChainSafe/forest/issues/4263

**Review by LesnyRumcajs — APPROVED** [review]

**Review by ruseinov — APPROVED** [review]

**Pull request merged**